action/protocol/execution/evm/evmstatedbadapter_erigon.go

```diff
@@ -0,0 +1,200 @@
+package evm
+
+import (
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/holiman/uint256"
+	erigonchain "github.com/ledgerwatch/erigon-lib/chain"
+	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	types2 "github.com/ledgerwatch/erigon-lib/types"
+	erigonstate "github.com/ledgerwatch/erigon/core/state"
+	"go.uber.org/zap"
+
+	"github.com/iotexproject/go-pkgs/hash"
+
+	"github.com/iotexproject/iotex-core/v2/blockchain/genesis"
+	"github.com/iotexproject/iotex-core/v2/pkg/log"
+	"github.com/iotexproject/iotex-core/v2/state"
+)
+
+type ErigonStateDBAdapter struct {
+	*StateDBAdapter
+	intra *erigonstate.IntraBlockState
+}
+
+type ErigonStateDBAdapterDryrun struct {
+	*ErigonStateDBAdapter
+}
+
+func NewErigonStateDBAdapter(adapter *StateDBAdapter,
+	intra *erigonstate.IntraBlockState,
+	chainRules *erigonchain.Rules,
+) *ErigonStateDBAdapter {
+	adapter.newContract = func(addr hash.Hash160, account *state.Account) (Contract, error) {
+		return newContractAdapter(addr, account, adapter.sm, intra, adapter.asyncContractTrie)
+	}
+	return &ErigonStateDBAdapter{
+		StateDBAdapter: adapter,
+		intra:          intra,
+	}
+}
+
+func NewErigonStateDBAdapterDryrun(adapter *StateDBAdapter,
+	intra *erigonstate.IntraBlockState,
+	chainRules *erigonchain.Rules,
+) *ErigonStateDBAdapterDryrun {
+	a := NewErigonStateDBAdapter(adapter, intra, chainRules)
+	adapter.newContract = func(addr hash.Hash160, account *state.Account) (Contract, error) {
+		return newContractErigon(addr, account, intra)
+	}
+	return &ErigonStateDBAdapterDryrun{
+		ErigonStateDBAdapter: a,
```

Comment:
what is the dryrunner used for? and it's difference with `ErigonStateDBAdapter`?

Reply:
dryrun is used for simulation, where reading is based on erigon instead of statedb